File: AD-HOC/1218/main.rb
def get_shoes(n)
  shoes = {"F" => 0, "M" => 0}
  line = gets.strip.split
  (0...line.length).step(2).each do |i|
    size, genre = line[i].to_i, line[i + 1]
    shoes[genre] += 1 if size == n
  end
  shoes
end

def formatted_print(c, shoes)
  puts if c > 1
  puts "Caso #{c}:"
  puts "Pares Iguais: #{shoes["F"] + shoes["M"]}"
  puts "F: #{shoes["F"]}"
  puts "M: #{shoes["M"]}"
end

$stdin.each.with_index(1) do |n, c|
  formatted_print(c, get_shoes(n.to_i))
end

File: AD-HOC/1218/main.py
from sys import stdin


def get_shoes():
    shoes = {}
    line = input().strip().split()
    while line:
        size, genre = line.pop(0), line.pop(0)
        if size not in shoes:
            shoes[size] = {'F': 0, 'M': 0}
        shoes[size][genre] += 1
    return shoes


def formatted_print(c, n, shoes):
    f = shoes[n]['F'] if n in shoes else 0
    m = shoes[n]['M'] if n in shoes else 0
    if c:
        print()
    print(f"Caso {c + 1}:")
    print(f"Pares Iguais: {f + m}")
    print(f"F: {f}")
    print(f"M: {m}")


for c, n in enumerate(map(str.strip, stdin)):
    shoes = get_shoes()
    formatted_print(c, n, shoes)
